pyro/infer/autoguide/guides.py

```diff
@@ -589,7 +591,9 @@ def get_posterior(self, *args, **kwargs):
         """
         Returns a diagonal Normal posterior distribution.
         """
-        return dist.Normal(self.loc, self.scale).to_event(1)
+        transform = dist.transforms.AffineTransform(self.loc, self.scale)
+        return dist.TransformedDistribution(dist.Normal(0., 1.).expand([self.latent_dim]).to_event(1),
+                                            transform)
```

Comment:
1. Please measure memory and time overhead introduced by this refactoring. This is a heavily used autoguide, and should be kept simple and fast. I am even inclined to introduce a new method `get_posterior_transformed_dist()` or something so as not to obfuscate this code.

2. Please take care with device (the proposed change disregards non-default device):
    ```py
    base_dist = dist.Normal(torch.zeros_like(self.loc), torch.ones_like(self.scale))
    return dist.TransformedDistribution(base_dist.to_event(1), transform)
    ```

Reply:
Reverted as discussed.